### ISSUE TITLE
Empty test cases report gas consumed

### ISSUE THREAD

**Issue by ldub** [cgewecke/eth-gas-reporter#81]
Consider the following diff creating an empty test case:

```diff
diff --git a/mock/test/metacoin.js b/mock/test/metacoin.js
index f7e6397..b146518 100644
--- a/mock/test/metacoin.js
+++ b/mock/test/metacoin.js
@@ -34,6 +34,7 @@ contract('MetaCoin', function (accounts) {
       assert.equal(metaCoinEthBalance, 2 * metaCoinBalance, 'Library function returned unexpected function, linkage may be broken')
     })
   })
+  it('should not consume any gas when an empty test case runs', function() {});
   it('should send coin correctly', function () {
     var meta
```

We should not expect this test case to report gas being consumed. However, it does:

```
  Contract: MetaCoin
    ✓ should put 10000 MetaCoin in the first account (347432 gas)
    ✓ should call a function that depends on a linked library (347432 gas)
    ✓ should not consume any gas when an empty test case runs (347432 gas)
    ✓ should send coin correctly (398425 gas)
```

You can also see that the previous test which only uses calls and doesn't send any transactions also reports gas usage even though the calls do not consume gas.

Tested on macOS 10.13.5, truffle v4.1.11, eth-gas-reporter v0.1.17 (and v0.1.8)

**Comment by cgewecke** [comment]
@ldub Is the test consuming gas in a `beforeEach` - by deploying something in the setup?

**Comment by cgewecke** [comment]
Yes it looks like this bug (#2) was re-introduced recently, possibly when we converted to `syncRequest`. . . will take a look and make sure we're setting the blockNumber in the correct mocha hook.

**Comment by ldub** [comment]
I did see that the 'hook end' is indeed properly incrementing the startBlock between tests. However without a more thorough understanding of mocha and eth-gas-reporter I can't be productive in digging deeper. Thanks a lot for looking.

**Comment by cgewecke** [comment]
@ldub Thanks for catching this! Really appreciate it. Should be fixed in `0.1.9`.

**Comment by ldub** [comment]
tested and works. thanks a lot for the quick fix :)
